### ISSUE TITLE
help!i need your help!!

### ISSUE THREAD

**Issue by userd171** [Ligeng-c/CVD_swin#2]
I am attempting to apply your methodology in my own research, but I have encountered some difficulties. When I run train.py, both your dataset and my newly assembled dataset exhibit an issue where black and white areas are inverted. Areas that were originally black in the original images become white in the generated images after training, and areas that were originally white become black. Below is a display of the image results from running your and my datasets, hence my somewhat presumptuous request.
![18870](https://github.com/user-attachments/assets/218ed7b0-bd6b-4e86-9884-29bd10af6c76)
![12000](https://github.com/user-attachments/assets/d1bf4258-123a-4966-84fd-8bde0d33d8aa)

Therefore, I would like to ask for your assistance with a couple of questions:

Could you provide an example of the trained model from your paper? This would greatly aid my understanding of the implementation details of the model.
Would it be possible for you to share the test.py file? Being somewhat limited in my abilities, I am in dire need of your guidance and assistance.
I apologize if these requests are an inconvenience to you. However, I believe that your help would significantly advance my research work. I look forward to your reply and appreciate your time in reading my email amidst your busy schedule.

**Comment by auto-Dog** [comment]
Have you solved your problem? I got similar results.

**Comment by auto-Dog** [comment]
> Have you solved your problem? I got similar results.

I think the problem exists because you did not finetune the base model. In the paper, the author train the base model first and does not use SSIM loss, which is the settings in readme. However, the acceptable results are finetuned with SSIM loss. You may set `--lambda_ssim 0.25`. 

What's more, I find the code have some issues when calculating SSIM. Normally SSIM are calculated on non-negative valued RGB images. But the code use [-1:1,0:1,0:1] Lab image for calculation. 

![Image](https://github.com/user-attachments/assets/42ab08cc-224d-4d2f-aab8-56d8a269139c)
